packages/ndk/lib/domain_layer/usecases/zaps/zaps.dart

```diff
@@ -141,11 +143,14 @@ class Zaps {
           final timeout = Timer(Duration(seconds: 30), () {
             _requests
                 .closeSubscription(zapResponse.zapReceiptResponse!.requestId);
+            if (zapResponse.streamSubscription!=null) {
+              zapResponse.streamSubscription!.cancel();
+            }
             Logger.log
                 .w("timed out waiting for zap receipt for invoice $invoice");
           });
 
-          zapResponse.zapReceiptResponse!.stream.listen((event) {
+          zapResponse.streamSubscription = zapResponse.zapReceiptResponse!.stream.listen((event) {
```

Comment:
why is the stream sub on the zapResponse?
As far as I can work it out its only used in zaps.dart, would be good to avoid side effects